parser.cpp

```diff
@@ -498,6 +498,7 @@ static const char *parse_num_full (const char *expr, int *value, int depth) {
 	int total = 0, last_num;
 	MATHOP last_op = M_NONE;
 	bool invert_lastnum, neg_lastnum;
+	const char *const expr_start = expr;

```

Comment:
That's by design. The value of `expr_start` should never be changed within the function, nor should what it points to.